src/gc/impl/conservative/gc.d

```diff
@@ -597,146 +597,123 @@ class ConservativeGC : GC
     private void *reallocNoSync(void *p, size_t size, ref uint bits, ref size_t alloc_size, const TypeInfo ti = null) nothrow
     {
         if (!size)
-        {   if (p)
-            {   freeNoSync(p);
-                p = null;
-            }
+        {
+            if (p)
+                freeNoSync(p);
             alloc_size = 0;
+            return null;
         }
-        else if (!p)
+        if (!p)
+            return mallocNoSync(size, bits, alloc_size, ti);
+
+        debug(PRINTF) printf("GC::realloc(p = %p, size = %llu)\n", p, cast(ulong)size);
+
+        Pool *pool = gcx.findPool(p);
+        if (!pool)
+            return null;
+
+        size_t psize;
+        size_t biti;
```

Comment:
I figure this anachronism (variable declaration way before initialization) was kept as a sort of a style background compatibility. Perhaps it's worth dropping that and moving the declarations as close to their initialization as possible.